src/wallet/api/swapService.ts

```
@@ -0,0 +1,79 @@
+  const aesEncryptedObject = AES.encrypt(serialized, password);
+
+  // Serializing the cipher output in Base64 to avoid loss of encryption parameter data
+  const baseEncryptedObj = CryptoJS.enc.Base64.parse(aesEncryptedObject.toString());
+  return CryptoJS.enc.Base64.stringify(baseEncryptedObj);
+}
+
+/**
+ * Decrypts a string ( PartialTx or Signatures ) from the backend
+ * @param serialized
+ * @param password
+ */
+export function decryptString(serialized: string, password: string): string {
+  const aesDecryptedObject = AES.decrypt(serialized, password);
+  return aesDecryptedObject.toString(CryptoJS.enc.Utf8);
+}
+
+/**
+ * Hashes the password to use it as an authentication on the backend
+ * @param {string} password
+ * @returns {string} hashed password
+ */
+export function hashPassword(password): string {
+  return sha256(password).toString();
+}
+
+/**
+ * Returns an axios instance pre-configured for interacting with the Atomic Swap Service
+ * @param [timeout] Optional timeout, defaults to the lib's timeout constant
+ * @param [network] Optional network. If not present, defaults connection to the lib's configured baseUrl
+ */
+const axiosInstance = async (timeout: number = TIMEOUT, network?: 'mainnet'|'testnet') => {
+  const swapServiceBaseUrl = config.getSwapServiceBaseUrl(network);
+  const defaultOptions = {
+    baseURL: swapServiceBaseUrl,
+    timeout: timeout,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
+
+  return axios.create(defaultOptions);
+};
+
+/**
+ * Calls the Atomic Swap Service requesting the creation of a new proposal identifier for the informed partialTx.
+ * @param serializedPartialTx
+ * @param password
+ */
+export const create = async (serializedPartialTx: string, password: string) => {
+  const swapAxios = await axiosInstance();
+  const payload = {
+    partialTx: encryptString(serializedPartialTx, password),
+    authPassword: hashPassword(password),
+  };
```

:speech_balloon: What are the possibilities in this request to go wrong? Should we handle a failure here or let it bubble up a possible error?

---

I think it's not a problem to let the eventual failures bubble up here, since they're usually related to `crypto-js` and could have valuable information on where/why the problem happened.

However, it has happened to me once to not receive the `serializedPartialTx` due to a bug and it got me some time to figure it out. Added a parameter validation to the function just in case.